File: Plugins/Aspose_Diagram_Java_for_Ruby/lib/asposediagramjava/Diagrams/removeallmacrosfromdiagram.rb
module Asposediagramjava
  module RemoveAllMacrosFromDiagram
    def initialize()
        data_dir = File.dirname(File.dirname(File.dirname(File.dirname(__FILE__)))) + '/data/'

        # Create instance of Diagram
        diagram = Rjb::import('com.aspose.diagram.Diagram').new(data_dir + "Drawing.vsd")

        # remove all macros
        diagram.setVbProjectData(nil)

        # Save as VDX
        diagram.save(data_dir + "RemoveAllMacros.vdx", Rjb::import('com.aspose.diagram.SaveFileFormat').VDX)

        puts "Removed all macros from diagram successfully!"
    end
  end
end

File: Plugins/Aspose_Diagram_Java_for_PHP/src/aspose/diagram/WorkingwithDiagrams/RemoveAllMacrosFromDiagram.php
<?php
namespace Aspose\Diagram\WorkingwithDiagrams;

use com\aspose\diagram\Diagram as Diagram;
use com\aspose\diagram\SaveFileFormat as SaveFileFormat;
class RemoveAllMacrosFromDiagram{

    public static function run($dataDir=null){

        # Create instance of Diagram
        $diagram = new Diagram($dataDir."drawing.vsd");

        # remove all macros
        $diagram->setVbProjectData(null);

        # Save as VDX
        $saveFileFormat=new SaveFileFormat();
        $diagram->save($dataDir."RemoveAllMacros.vdx", $saveFileFormat->VDX);
        print "Removed all macros from diagram successfully!".PHP_EOL;
    }
}